utils.go

```diff
@@ -78,3 +78,19 @@ func cleanupFds(fds []*os.File, numFds int) {
 		_ = fds[i].Close()
 	}
 }
+
+// writeToFile opens a file in write only mode and writes bytes to it
+func writeToFile(path string, data []byte) error {
+	f, err := os.OpenFile(path, os.O_WRONLY, 0755)
```

Comment:
You might want to define a const `fileMode0755` that could be shared across the entire package.